### PR TITLE
Fixing __repr__ function in array

### PR DESCRIPTION
**Pull request by abdulasiraj**
closes https://trello.com/c/SH9tX7ni

### CODE REVIEW

File: ivy/data_classes/array/array.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 import numpy as np
 from operator import mul
 from typing import Optional
+from time import perf_counter
```
Comment (VedPatwardhan) [inline ivy/data_classes/array/array.py:8]: Probably an import left over from debugging, could you please remove it?

File: ivy/data_classes/array/array.py [hunk 2]
```diff
@@ -370,19 +372,19 @@ def __repr__(self):
...
                 + repr[repr.find("(") :]
                 + self._post_repr.format(ivy.current_backend_str())
             )
-
+        
     def __dir__(self):
         return self._data.__dir__()
 
     def __getattribute__(self, item):
+        # print('in the get attr function')
```
Comment (VedPatwardhan) [inline ivy/data_classes/array/array.py:387]: could you please remove this comment too?

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hey @abdulasiraj, thanks for taking on this task! Just requested a few changes 🙂

File: ivy/data_classes/array/array.py [hunk 3]
```diff
@@ -370,19 +372,19 @@ def __repr__(self):
             # from the currently set backend
             backend = ivy.with_backend(self.backend, cached=True)
         arr_np = backend.to_numpy(self._data)
-        rep = ivy.vec_sig_fig(arr_np, sig_fig) if self.size > 0 else np.array(arr_np)
```
[the remark this answers is not in the document]
Comment (abdulasiraj) [reply ivy/data_classes/array/array.py:373]: these lines can be removed safely.

File: ivy/data_classes/array/array.py [hunk 4]
```diff
@@ -363,8 +364,7 @@ def __repr__(self):
                 self._post_repr = ", dev={})".format(self._dev_str)
             else:
                 self._post_repr = ")"
-        sig_fig = ivy.array_significant_figures
-        dec_vals = ivy.array_decimal_values
+        dec_vals = ivy.array_decimal_values()
```
Comment (VedPatwardhan) [inline ivy/data_classes/array/array.py:367]: please revert the change to `dec_vals` as `ivy.array_decimal_values` isn't a function

File: ivy/data_classes/array/array.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (VedPatwardhan) [reply ivy/data_classes/array/array.py:376]: Actually if you check the code, `ivy.array_significant_figures` allows users to decide the number of significant figures they'd like to be concerned about. Based on the number of significant figures defined by the user, we would be able to control the number of significant figures. Removing the use of `ivy.vec_sig_fig` loses that control.
For e.g. the below code returns the following result without the changes,
```
import ivy

ivy.set_array_significant_figures(5)
x = ivy.array([1.23124353453634634636])
print(x)
ivy.set_array_significant_figures(10)
print(x)
```
```
ivy.array([1.2312])
ivy.array([1.23124349])
```
while with the changes, it's the following,
```
ivy.array([1.2312435])
ivy.array([1.2312435])
```
We should aim to optimize the `__repr__` method of course, but we'd also want to retain that control 😅

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hey @abdulasiraj, just suggested a few more changes, thanks 🙂

File: ivy/data_classes/array/array.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (abdulasiraj) [reply ivy/data_classes/array/array.py:376]: Oh, my bad. Thanks, I'll look into this.

**Comment by abdulasiraj** [comment]
Hi @VedPatwardhan.

This PR is ready to review. Slow down occurs due to the expensive computation of significant figures. With the help @kgmann and [this](https://stackoverflow.com/questions/18915378/rounding-to-significant-figures-in-numpy) thread on stackoverflow. Things are now under controlled.

File: ivy/__init__.py [hunk 5]
```diff
@@ -843,24 +843,18 @@ def _assert_array_significant_figures_formatting(sig_figs):
 
 
 # ToDo: SF formating for complex number
-def _sf(x, sig_fig=3):
+def sf(x, sig_fig=3):
```
Comment (vedpatwardhan) [inline ivy/__init__.py:846]: let's rename this function to something verbose for code clarity, probably just `vec_sig_fig`

File: ivy/data_classes/array/array.py [hunk 6]
```diff
@@ -6,6 +6,7 @@
 from operator import mul
 from typing import Optional
 
+
```
Comment (vedpatwardhan) [inline ivy/data_classes/array/array.py:9]: extra newline

File: ivy/data_classes/array/array.py [hunk 7]
```diff
@@ -1156,4 +1157,4 @@ def __iter__(self):
                 return iter([to_ivy(i) for i in ivy.unstack(self._data)])
             elif self.ndim == 1:
                 return iter([to_ivy(i).squeeze(0) for i in self._data])
-        return iter([to_ivy(i) for i in self._data])
+        return iter([to_ivy(i) for i in self._data])
```
Comment (vedpatwardhan) [inline ivy/data_classes/array/array.py:1160]: missing newline here

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Hey @abdulasiraj, thanks for making the changes, just a couple more changes with some formatting issues that can be fixed using `pre-commit` 🙂

**Comment by vedpatwardhan** [comment]
ivy-gardener

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
> Actually if you check the code, `ivy.array_significant_figures` allows users to decide the number of significant figures they'd like to be concerned about. Based on the number of significant figures defined by the user, we would be able to control the number of significant figures. Removing the use of `ivy.vec_sig_fig` loses that control. For e.g. the below code returns the following result without the changes,
> 
> ```
> import ivy
> 
> ivy.set_array_significant_figures(5)
> x = ivy.array([1.23124353453634634636])
> print(x)
> ivy.set_array_significant_figures(10)
> print(x)
> ```
> 
> ```
> ivy.array([1.2312])
> ivy.array([1.23124349])
> ```
> 
> while with the changes, it's the following,
> 
> ```
> ivy.array([1.2312435])
> ivy.array([1.2312435])
> ```
> 
> We should aim to optimize the `__repr__` method of course, but we'd also want to retain that control 😅

The changes made don't seem to break anything, but the example I shared previously still doesn't seem to work properly with the changes, probably there's some fixes to be made to the logic. Thanks @abdulasiraj 🙂

File: ivy/data_classes/array/array.py [hunk 8]
```diff
@@ -373,9 +373,9 @@ def __repr__(self):
             # from the currently set backend
             backend = ivy.with_backend(self.backend, cached=True)
         arr_np = backend.to_numpy(self._data)
-        rep = ivy.vec_sig_fig(arr_np, sig_fig) if self.size > 0 else np.array(arr_np)
+        arr_np = ivy.vec_sig_fig(arr_np, sig_fig) if self.size > 0 else np.array(arr_np)
```
Comment (vedpatwardhan) [inline ivy/data_classes/array/array.py:376]: we could probably rename this back to `rep`

**Review by vedpatwardhan — APPROVED** [review]
One last change, feel free to merge the PR once done. Thanks @abdulasiraj 🙂

**Pull request merged**